### DISCUSSION TITLE
getInitialProps query is empty after export. works on dev

### DISCUSSION THREAD

**Discussion by kashkashio** [vercel/next.js#22476] [Help] [answered]
Hey,
I'm running next app version 10.0.0.

my scenario is this:
I'm trying to fetch page meta data like title and open graph stuff before page loads.
To do so i am using the function "getInitialProps" which actually works perfect on development run..

But after deploying my app to aws amplify, the query becomes empty and it dosn't works anymore.

I've did some research and found that using "getServerSideProps" can do the trick, so i tried that and it worked on localhost dev ! but when i tried to run a build i got the error that it cannot be exported and that i should remove "export" from build npm command which is impossible to do because aws amplify needs the "out" directory...

Any idea of how to achieve it?

I Just need to be able to get the query param and send it in http request before the page rendres..

Thank you !

**Comment by indegser** [comment, marked as answer]
It seems query params are dynamic, thus cannot be generated at build time.
If you must use aws amplify(system with public directory and no server/serverless support)
fetching open graph in the client side is the only solution.

1. Create open graph fetching serverless function with aws lamda or a server with aws ec2
2. When your app runs, send request to above, while suspending page rendering (like displaying spinner)

Or run your own custom server with 'next start' in aws ec2 or something.

**Reply by kashkashio** [reply]
Thanks, i was afraid that these are my only options.
Went all in with running it independently on ec2 behind reverse porxy of nginx... running build than start and it works like a charm !

Thanks tho !
